docs/self-hosted-runner/SelfHostedRunner.md

```diff
@@ -0,0 +1,143 @@
+# arduino-ide 2.0.0 for Raspberry Pi
+
+These are instructions on compiling [arduino-ide 2](https://github.com/arduino/arduino-ide) for arm64 using self-hosted runners. Compiling can be done on raspberry pi os 64-bit or ubuntu arm64. The compiled binaries run on raspberry pi os 64-bit and ubuntu arm64.
+
+Compiling is done using github actions, the same build process as on X86.
+
+Because github does not offer linux arm64 runners, compilation is done using a self-hosted runner on a raspberry pi. 
+
+Because github does not recommend running self-hosted runners on public repositories, the arduino-ide repository is forked first, and public PR are switched off on the fork.
+
+For repeatability and security, compilation is done inside a docker.
```

Comment:
This approach could be enhanced, IMHO. Having a dedicated runner is a problem:
- we have to manually operate on it and update it regularly 
- it's a security risk
- I don't think that having a forked private version only for the self-hosted runner will pay off in the future
- 
Maybe using this amazing piece of technology would be better:
https://github.com/docker/setup-qemu-action
Basically, this allows to run docker containers built for another architecture directly inside the runners provided by GitHub. Like this one for example: https://hub.docker.com/r/amd64/ubuntu/

We use the same approach to build [imgtool](https://github.com/mcu-tools/mcuboot/tree/main/scripts)
https://github.com/arduino/imgtool-packing/blob/bcb0ac50d94633784c51a0833fe2346e8cf8b2aa/.github/workflows/release.yml#L118-L187